Source/WebCore/platform/graphics/coretext/DrawGlyphsRecorderCoreText.cpp

```diff
@@ -237,16 +237,56 @@ void DrawGlyphsRecorder::recordEndLayer(CGRenderingStateRef, CGGStateRef gstate)
     m_owner.endTransparencyLayer();
 }
 
-static Vector<CGSize> computeAdvancesFromPositions(const CGPoint positions[], size_t count, const CGAffineTransform& textMatrix)
+struct AdvancesAndInitialPosition {
+    Vector<CGSize> advances;
+    CGPoint initialPosition;
+};
+
+static AdvancesAndInitialPosition computeHorizontalAdvancesFromPositions(const CGPoint positions[], size_t count, const CGAffineTransform& textMatrix)
 {
-    Vector<CGSize> result;
+    // This function needs to be the inverse of fillVectorWithHorizontalGlyphPositions().
+
+    ASSERT(count); // Because we say "positions[0]" below.
+
+    AdvancesAndInitialPosition result;
+    result.advances.reserveInitialCapacity(count);
+    result.initialPosition = CGPointApplyAffineTransform(positions[0], textMatrix);
     for (size_t i = 0; i < count - 1; ++i) {
         auto nextPosition = positions[i + 1];
         auto currentPosition = positions[i];
         auto advance = CGSizeMake(nextPosition.x - currentPosition.x, nextPosition.y - currentPosition.y);
-        result.append(CGSizeApplyAffineTransform(advance, textMatrix));
+        result.advances.uncheckedAppend(CGSizeApplyAffineTransform(advance, textMatrix));
     }
-    result.constructAndAppend(CGSizeMake(0, 0));
+    result.advances.uncheckedConstructAndAppend(CGSizeZero);
+    return result;
+}
+
+static AdvancesAndInitialPosition computeVerticalAdvancesFromPositions(const CGSize translations[], const CGPoint positions[], unsigned count, float ascentDelta, AffineTransform textMatrix)
+{
+    // This function needs to be the inverse of fillVectorWithVerticalGlyphPositions().
+
+    ASSERT(count); // Because we say "positions[0]" below.
+
+    auto syntheticObliqueLessTextMatrix = computeBaseVerticalTextMatrix(computeBaseOverallTextMatrix(std::nullopt)); // See fillVectorWithVerticalGlyphPositions(), which describes what this is.
```

Comment:
Not super fond of "oblique less" as a way to write "without oblique". No sure that "obliqueless" as a single word would be better. Maybe, "omitting oblique", so, `syntheticTextMatrixOmittingOblique`?